### PR TITLE
Update CI as a precursor to enabling merge queues

### PR DESCRIPTION
**Pull request by urschrei**
See #142

- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `rstar/CHANGELOG.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,26 +1,41 @@
...
     needs:
       - rstar
       - check
       - no_std
+    if: success()
     steps:
       - name: Mark the job as a success
-        if: success()
         run: exit 0
+  ci-failure:
```
Comment (adamreichold) [inline .github/workflows/test.yml:29]: It is somewhat reasonable, but just to have the alternative discussed here, we could also have a single job dynamically checking the other jobs results, e.g. in PyO3 we use

```yaml
conclusion:
needs:
    - fmt
    - check-msrv
    - clippy
    - build-pr
    - build-full
    - valgrind
    - careful
    - coverage
    - emscripten
if: always()
runs-on: ubuntu-latest
steps:
    - name: Result
    run: |
        jq -C <<< "${needs}"
        # Check if all needs were successful or skipped.
        "$(jq -r 'all(.result as $result | (["success", "skipped"] | contains([$result])))' <<< "${needs}")"
    env:
        needs: ${{ toJson(needs) }}
```

Comment (adamreichold) [reply .github/workflows/test.yml:29]: Alternatively, we could at least factor out the job list using a YAML anchor?

Comment (adamreichold) [reply .github/workflows/test.yml]: Certainly more DRY.

**Review by adamreichold — APPROVED** [review]

**Comment by adamreichold** [comment]
Did you configure the merge queue for squash merging? Otherwise, I would suggest manually squashing this PR before merging.

**Comment by urschrei** [comment]
> Did you configure the merge queue for squash merging? Otherwise, I would suggest manually squashing this PR before merging.

Configured to "squash and merge"

**Comment by urschrei** [comment]
Hum. That shouldn't have merged before the tests passed

**Comment by urschrei** [comment]
> Hum. That shouldn't have merged before the tests passed

Ah forgot to update setting with new ci check name. Now needs `conclusion` to pass before merging.

**Pull request merged**